app/components/login-form.js

```diff
@@ -0,0 +1,43 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+export default class LoginFormComponent extends Component {
+    @service router;
+    @tracked emailError;
+    @tracked passwordError;
+    @tracked email;
+    @tracked password;
+
+    validateEmail(email) {
+        const emailPattern = /^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+        return emailPattern.test(email);
+    }
+
+    @action
+    async signIn(event) {
+        event.preventDefault();
+
+        console.log(this.email);
+        console.log(this.password);
+
+        if (!this.email || !this.validateEmail(this.email)) {
```

Comment:
Can you check below scenario?
email Id blank
password given
click submit...navigation should not happen,
may i know whether this scenario covers ur change.